DigitalLearningSolutions.Data.Migrations/202011101344_CompetencyFrameworkDBChanges.cs

```diff
@@ -0,0 +1,172 @@
+                .WithColumn("CreatedDate").AsDateTime().NotNullable().WithDefaultValue(SystemMethods.CurrentDateTime)
+                .WithColumn("BrandID").AsInt32().NotNullable().ForeignKey("Brands", "BrandID").WithDefaultValue(6)
+                .WithColumn("CategoryID").AsInt32().NotNullable().ForeignKey("CourseCategories", "CourseCategoryID").WithDefaultValue(1)
+                .WithColumn("TopicID").AsInt32().NotNullable().ForeignKey("CourseTopics", "CourseTopicID").WithDefaultValue(1)
+                .WithColumn("ParentRoleProfileID").AsInt32().Nullable().ForeignKey("RoleProfiles", "ID")
+                .WithColumn("National").AsBoolean().NotNullable().WithDefaultValue(false)
+                .WithColumn("Public").AsBoolean().NotNullable().WithDefaultValue(false)
+                .WithColumn("OwnerAdminID").AsInt32().NotNullable().ForeignKey("AdminUsers", "AdminID")
+                .WithColumn("Archived").AsDateTime().Nullable()
+                .WithColumn("LastEdit").AsDateTime().NotNullable().WithDefaultValue(SystemMethods.CurrentDateTime);
+            //DLSV2-45 New RoleProfileRequirements table
+            Create.Table("RoleProfileRequirements")
+              .WithColumn("ID").AsInt32().NotNullable().PrimaryKey().Identity()
+              .WithColumn("RoleProfileID").AsInt32().Nullable().ForeignKey("RoleProfiles", "ID")
+              .WithColumn("CompetencyID").AsInt32().NotNullable().ForeignKey("Competencies", "ID");
+            //DLSV2-97 Add OrderNumber int field to SelfAssessmentStructure table
+            Alter.Table("SelfAssessmentStructure")
+               .AddColumn("Ordering").AsInt32().NotNullable().WithDefaultValue(1);
+            //DLSV2-99 Additional fields to SelfAssessments table:
+            Alter.Table("SelfAssessments")
+                .AddColumn("SubmittedForSignOff").AsDateTime().Nullable()
+                .AddColumn("OutcomeId").AsInt32().Nullable()
+                .AddColumn("SignedByAdminId").AsInt32().Nullable().ForeignKey("AdminUsers", "AdminID")
+                .AddColumn("Signed").AsDateTime().Nullable()
+                .AddColumn("OutcomeAgreed").AsBoolean().NotNullable().WithDefaultValue(false);
+            //DLSV2-98 Additional fields to SelfAssessmentResults table:
+            Alter.Table("SelfAssessmentResults")
+                .AddColumn("SupportingComments").AsString(int.MaxValue).Nullable()
+                .AddColumn("VerificationRequested").AsDateTime().Nullable()
+                .AddColumn("VerificationOutcomeId").AsInt32().Nullable()
+                .AddColumn("VerifierAdminID").AsInt32().Nullable().ForeignKey("AdminUsers", "AdminID")
+                .AddColumn("VerifierComments").AsString(int.MaxValue).Nullable()
+                .AddColumn("VerifiedDate").AsDateTime().Nullable();
+            Execute.Sql(Properties.Resources.DLSV2_95_AddSystemVersioning);
+        }
+        public override void Down()
```

Comment:
Have you checked that this `Down` migration works? Worth double checking, just in case we need to undo the migration at any point. I think some of the existing `Down` migrations might not work at the moment, at least I was having some issues with them 🙂

Reply:
will do 👍

Reply:
The Down() migration works, now (after minor headaches!) :)